utils/streams.js

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env node
+
+const program = require('commander');
+const through = require('through2');
+const util = require('./utils');
+const fs = require('fs');
+const Converter = require('csvtojson').Converter;
+const csvConverter = new Converter({
+    constructResult: false,
+    toArrayString: true
+});
+const {promisify} = require('util');
+const readdir = promisify(fs.readdir);
+const appendFile = promisify(fs.appendFile);
+const readFile = promisify(fs.readFile);
+
+const resultPath = './data/css/​bundle.css';
+const appendixPath = './data/css/nodejs-homework3.css';
+
+
+function reverse(str) {
+    str
+        .pipe(through(function (buffer, encodding, done) {
+            this.push(util.reverse(buffer));
+            done();
+            process.exit(0);
+        }))
+        .pipe(process.stdout);
+}
+
+function transform(str) {
+    str
+        .pipe(through(function (buffer, encodding, done) {
+            this.push(util.toUpperCase(buffer));
+            done();
+            process.exit(0);
+        }))
+        .pipe(process.stdout);
+}
+
+function outputFile(filePath) {
+    if (!filePath) program.help();
+    fs.createReadStream(filePath)
+        .on('error', function () {
+            console.error('File not found: ' + filePath);
+            program.help();
+        })
+        .pipe(process.stdout);
+}
+
+function convertFromFile(filePath) {
+    fs.createReadStream(filePath)
+        .on('error', function () {
+            console.error('File not found: ' + filePath);
+            program.help();
+        })
+        .pipe(csvConverter)
+        .pipe(process.stdout);
+}
+
+function convertToFile(filePath) {
+    fs.createReadStream(filePath)
+        .on('error', function () {
+            console.error('File not found: ' + filePath);
+            program.help();
+        })
+        .pipe(csvConverter)
+        .pipe(fs.createWriteStream(filePath.substring(0, filePath.lastIndexOf('.')) + '.json'));
+}
+
+function cssBundler(filePath) {
+    console.log(filePath);
+    readdir(filePath)
+        .then(files => {
+            files.forEach(fileName => {
+                appendToResult(filePath + '/' + fileName);
```

Comment:
Async operation in sync loop.

Possible way to resolve it:
```javascript
.then(files => {
    return files.map(fileName => appendToResult(fileName))   // .map return new array of promises
}
.then
```

Reply:
But to use this approach (it's called promise chaining) the function `appendToResult` should be async